lib/collection/src/collection/mod.rs

```diff
@@ -442,8 +449,14 @@ impl Collection {
     ) -> CollectionResult<()> {
         // Check for disabled replicas
         let shard_holder = self.shards_holder.read().await;
+        let get_shard_transfer = |key| shard_holder.get_transfer(&key);
         for replica_set in shard_holder.all_shards() {
-            replica_set.sync_local_state().await?;
+            // TODO:
+            //   Aborting shard transfer here doesn't *really* work, if we keep requesting new
+            //   transfer in `Collection::request_shard_transfer` (at the bottom).
+            //
+            //  Shard transfer should be initiated by the *sender*.
+            replica_set.sync_local_state(get_shard_transfer).await?;
```

Comment:
- we abort shard transfer, if transfer-receiver detects that transfer-sender failed